### PR TITLE
feat: perfomance and refactoring changes

### PR DESCRIPTION
**Pull request by tricantivu**
Hi!

I noticed the script runs some commands in subshells, those include `apt` and `expr`. That that may be a cause of slowdowns so I made some [research](https://stackoverflow.com/a/48016366) and foundnew approaches that are more efficient.

Also, I believe using `type` builtin command is better than `command` since the latter *shows descriptions* and the former *paths to executables*.

Cheers!

### CODE REVIEW

File: rhino-deinst [hunk 1]
```diff
@@ -1,12 +1,18 @@
 #!/bin/bash
 set -e
 
-if ! command -v dialog > /dev/null; then
```
Comment (Elsie19) [inline rhino-deinst]: None of these are subshells. A subshell would be something like `( echo "hello" )`. Brackets `{}` are not subshells, and don't inherently slow down a program. On top of that, performance is not the goal of this program. It doesn't need to be lightning fast.

**Review by Elsie19 — CHANGES_REQUESTED** [review]

File: rhino-deinst [hunk 2]
```diff
@@ -163,16 +163,17 @@ echo "Updating package cache..."
 sudo apt-get update
 
 echo "Checking for package existence..."
+
 for i in "${desktop[@]}"; do
-  if [[ $(apt-cache search --names-only "^$i\$") ]]; then
```
Comment (Elsie19) [inline rhino-deinst]: `$()` is a shell substitution, not a subshell

**Review by Elsie19 — CHANGES_REQUESTED** [review]

File: rhino-deinst (hunk 1 above)
Comment (tricantivu) [reply rhino-deinst]: I meant the `$( expr ... )` and `$( apt ...)` lines.

File: rhino-deinst (hunk 2 above)
Comment (tricantivu) [reply rhino-deinst]: It's command substitution and it runs on a subshell (https://wiki.bash-hackers.org/syntax/ccmd/grouping_subshell)  (http://mywiki.wooledge.org/BashGuide/CompoundCommands)

Quoting Greg's guide:


> The following commands create subshells:
>
> (cmd)
> cmd1 | cmd2
> $(cmd) or `cmd`
>
> cmd &
>
> <(cmd) or >(cmd)
>

**Comment by tricantivu** [comment]
I don't if it has been noticed but changes have been made.

**Comment by Elsie19** [comment]
Oh sorry, forgot about this PR

File: rhino-deinst [hunk 3]
```diff
@@ -6,7 +6,11 @@ if ! command -v dialog > /dev/null; then
   sudo apt-get install dialog -y
 fi
 
-SELECTED=($( dialog --backtitle "Press 'space' to toggle an option, use arrow keys to go up/down, press enter to continue" --scrollbar --separate-output --checklist "Select Desktop Environment(s)" $(expr $(tput lines) - 5) $(expr $(tput cols) - 5) 5 \
+# Set LINES and COLUMNS variables. (Reference: https://stackoverflow.com/a/48016366)
+shopt -s checkwinsize
+cat /dev/null
```
Comment (Elsie19) [inline rhino-deinst:11]: what's this for?

Comment (tricantivu) [reply rhino-deinst:11]: To not use tput and command substitution to manipulate the terminal's length and columns.

**Review by Elsie19 — APPROVED** [review]

**Pull request merged**